parsing/location.ml

```diff
@@ -223,9 +223,29 @@ let rec highlight_locations ppf locs =
 
 open Format
 
+let rewrite_absolute_path =
+  let init = ref false in
+  let map_cache = ref None in
+  fun path ->
+    if not !init then begin
+      init := true;
+      match Sys.getenv "BUILD_PATH_PREFIX_MAP" with
```

Comment:
You should probably emit a warning (or at least some kind of informational message) when the environment variable is found.

Reply:
This is an interesting idea but I don't completely get it.

The idea is to make it possible for someone to check that the behavior correctly took that variable into account? That seems sensible.

Am I correct that we don't have anything existing in the compiler right now for "informational messages which are not warnings"? (That is, to factually let the user know about something, without suggesting that they are guilty of some offense.) How would you suggest to go about it?

Reply:
The idea is to let me known if I have this variable defined in my environment and I forgot about it. But you're right that it doesn't fit in any of the classes of message that the compiler currently outputs.

Reply:
@damiendoligez I have thought about this again, but I don't see an easy way to do it.

It sounds like a *lot* of work to implement (1) a new way for the compiler to print information about its execution (Information, so having an Error/Warning/Information continuum) and (2) user-interface mechanism to enable/disable printing of this new kind of messages.

I think your general wish is to have a way to observe the reliance of the compiler process over its execution environment. We could implement this as a new compiler option (`-trace-environment-access`), but note that there are third-party tools that already provide that monitoring:

```shell
$ echo "string secure_getenv(string)" > ltrace.conf
$ OCAML_COLOR=auto ltrace -F ./ltrace.conf -e secure_getenv -- ocamlc -c foo.ml
ocamlc->secure_getenv("OCAMLRUNPARAM")                 = nil
ocamlc->secure_getenv("CAMLRUNPARAM")                  = nil
ocamlc->secure_getenv("OCAMLRUNPARAM")                 = nil
ocamlc->secure_getenv("CAMLRUNPARAM")                  = nil
ocamlc->secure_getenv("TMPDIR")                        = nil
ocamlc->secure_getenv("TEMP")                          = nil
ocamlc->secure_getenv("OCAMLLIB")                      = nil
ocamlc->secure_getenv("CAMLLIB")                       = nil
ocamlc->secure_getenv("OCAML_BINANNOT_WITHENV")        = nil
ocamlc->secure_getenv("OCAMLPARAM")                    = nil
ocamlc->secure_getenv("OCAML_COLOR")                   = "auto"
ocamlc->secure_getenv("OCAMLPARAM")                    = nil
ocamlc->secure_getenv("OCAMLPARAM")                    = nil
```